packages/camera/camera_android_camerax/lib/src/aspect_ratio_strategy.dart

```diff
@@ -0,0 +1,138 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/services.dart';
+import 'package:meta/meta.dart' show immutable;
+
+import 'camerax_library.g.dart';
+import 'instance_manager.dart';
+import 'java_object.dart';
+
+/// The aspect ratio of a UseCase.
+///
+/// Aspect ratio is the ratio of width to height.
+///
+/// See https://developer.android.com/reference/androidx/camera/core/AspectRatio.
+class AspectRatio {
+  AspectRatio._();
+
+  /// 4:3 standard aspect ratio.
+  ///
+  /// See https://developer.android.com/reference/androidx/camera/core/AspectRatio#RATIO_4_3().
+  static const int ratio4To3 = 0;
+
+  /// 16:9 standard aspect ratio.
+  ///
+  /// See https://developer.android.com/reference/androidx/camera/core/AspectRatio#RATIO_16_9().
+  static const int ratio16To9 = 1;
+
+  /// The aspect ratio representing no preference for aspect ratio.
+  ///
+  /// See https://developer.android.com/reference/androidx/camera/core/AspectRatio#RATIO_DEFAULT().
+  static const int ratioDefault = -1;
+}
+
+/// The aspect ratio strategy defines the sequence of aspect ratios that are
+/// used to select the best size for a particular image.
+///
+/// See https://developer.android.com/reference/androidx/camera/core/resolutionselector/AspectRatioStrategy.
+@immutable
+class AspectRatioStrategy extends JavaObject {
+  /// Construct a [AspectRatioStrategy].
+  AspectRatioStrategy({
+    required this.preferredAspectRatio,
+    required this.fallbackRule,
+    super.binaryMessenger,
+    super.instanceManager,
+  })  : _api = _AspectRatioStrategyHostApiImpl(
+          instanceManager: instanceManager,
+          binaryMessenger: binaryMessenger,
+        ),
+        super.detached() {
+    _api.createFromInstances(this, preferredAspectRatio, fallbackRule);
+  }
+
+  /// Instantiates a [AspectRatioStrategy] without creating and attaching to an
+  /// instance of the associated native class.
+  ///
+  /// This should only be used outside of tests by subclasses created by this
+  /// library or to create a copy for an [InstanceManager].
+  AspectRatioStrategy.detached({
+    required this.preferredAspectRatio,
+    required this.fallbackRule,
+    super.binaryMessenger,
+    super.instanceManager,
+  })  : _api = _AspectRatioStrategyHostApiImpl(
+          instanceManager: instanceManager,
+          binaryMessenger: binaryMessenger,
+        ),
+        super.detached();
+
+  /// CameraX doesn't fall back to select sizes of any other aspect ratio when
+  /// this fallback rule is used.
+  ///
+  /// See https://developer.android.com/reference/androidx/camera/core/resolutionselector/AspectRatioStrategy#FALLBACK_RULE_NONE().
+  static const int fallbackRuleNone = 0;
```

Comment:
My initial thought is that changing them to getters wouldn't be a significant improvement. If the values change, we likely will be quick to find out. Plus, there's the risk that they change the name and a getter would fail there. Not sure which case is more likely but I think it's something to consider. @bparrishMines any thoughts on this?

Reply:
It's hard to change them to getters since it would require an async call. Which would make it unintuitive to use. You could also add a Java unit test that verifies that these constants stay the same. However, this would still be susceptible to an instance where the constant values are different for different Android versions. Although this case is probably extremely rare.